# HG changeset patch
# User Rik <rik@octave.org>
# Date 1452733916 28800
#      Wed Jan 13 17:11:56 2016 -0800
# Node ID 31d41418fa579872a7bf5ad82462a3fd076dd554
# Parent  73ab962bc52dcb091b2ad0ef058a816b95cdfc8f
build: Restore dependency of docs on example code.

* doc/module.mk: Add dependencies for oop.texi and external.texi on
$(examples_code_SRC).  Remove $(examples_code_SRC) from octave_TEXINFOS
declaration.  Remove $(munge_texi_SOURCES) which no longer expands to anything.

diff --git a/doc/module.mk b/doc/module.mk
--- a/doc/module.mk
+++ b/doc/module.mk
@@ -227,23 +227,18 @@ TXI_SRC = $(MUNGED_TEXI_SRC:.texi=.txi)
 BUILT_OCTAVE_TEXI_SRC = \
   doc/interpreter/contributors.texi \
   $(GRAPH_PROP_TEXI_SRC) \
   $(MUNGED_TEXI_SRC)
 
 info_TEXINFOS += \
   doc/interpreter/octave.texi
 
-## The $(examples_code_SRC) files are included here because
-## they are included in the manual via the @EXAMPLEFILE macro,
-## so they are dependencies for the documentation.
-
 octave_TEXINFOS = \
-  $(BUILT_OCTAVE_TEXI_SRC) \
-  $(examples_code_SRC)
+  $(BUILT_OCTAVE_TEXI_SRC)
 
 ## As of version 1.14.1, automake does not seem to generate
 ## rules for DVI, PDF, or HTML output that work for us when
 ## there are additional dependencies, so we include our own
 ## versions of the rules here.
 
 OCTAVE_HTML_DIR = doc/interpreter/octave.html
 OCTAVE_HTML_TMP_DIR = $(OCTAVE_HTML_DIR:.html=.htp)
@@ -336,17 +331,22 @@ octetc_DATA += \
   doc/interpreter/doc-cache \
   doc/interpreter/macros.texi
 
 doc/interpreter/doc-cache: $(DOCSTRING_FILES) doc/interpreter/mk_doc_cache.m | $(OCTAVE_INTERPRETER_TARGETS) doc/interpreter/$(octave_dirstamp)
 	$(AM_V_GEN)rm -f $@-t $@ && \
 	$(top_builddir)/run-octave -f -q -H $(srcdir)/doc/interpreter/mk_doc_cache.m - $(srcdir)/doc/interpreter/macros.texi $(DOCSTRING_FILES) > $@-t && \
 	mv $@-t $@
 
-$(MUNGED_TEXI_SRC): $(DOCSTRING_FILES) $(munge_texi_SOURCES)
+$(MUNGED_TEXI_SRC): $(DOCSTRING_FILES)
+
+## These two texi files have an additional dependency through the
+## @EXAMPLEFILE macro.
+doc/interpreter/oop.texi: $(examples_code_SRC)
+doc/interpreter/external.texi: $(examples_code_SRC)
 
 %.texi : %.txi doc/interpreter/munge-texi.pl | doc/interpreter/$(octave_dirstamp)
 	$(AM_V_GEN)rm -f $@-t $@ && \
 	$(PERL) $(srcdir)/doc/interpreter/munge-texi.pl $(top_srcdir) $(DOCSTRING_FILES) < $< > $@-t && \
 	mv $@-t $@
 
 doc/interpreter/contributors.texi: doc/interpreter/contributors.in | doc/interpreter/$(octave_dirstamp)
 	$(AM_V_GEN)rm -f $@-t $@ && \
